kani-compiler/src/kani_middle/transform/body.rs

```diff
@@ -428,6 +428,15 @@ impl MutableBody {
         self.blocks.push(BasicBlock { statements: Vec::default(), terminator })
     }
 
+    /// Replace statements from the given basic block
+    pub fn replace_statements(
+        &mut self,
+        source_instruction: &SourceInstruction,
```

Comment:
Please pass the basic block instead. The source instruction is that the function adjusts the instruction so it still points to the same instruction as before.

This code is replacing the basic block code which invalidates the source instruction.